foxpuppet/windows.py

```diff
@@ -2,44 +2,75 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this file,
 # You can obtain one at http://mozilla.org/MPL/2.0/.
 
-from contextlib import contextmanager
-
 from selenium.webdriver.common.by import By
 from selenium.webdriver.support.wait import WebDriverWait
 
 
-class Windows(object):
+class WindowManager(object):
 
     def __init__(self, selenium):
         self.selenium = selenium
-        self._handle = None
 
     @property
-    def all(self):
-        """Returns all current window handles"""
-        return self.selenium.window_handles
+    def windows(self):
+        return [BaseWindow(self.selenium, handle=handle)
```

Comment:
We only support Browser windows at the moment. Why should we return BaseWindow instances?